src/apps/backend/modules/account/account_service.py

```diff
@@ -12,13 +14,26 @@
 )
 from modules.authentication.authentication_service import AuthenticationService
 from modules.authentication.types import CreateOTPParams
+from modules.notification.types import NotificationPreferencesParams
 
 
 class AccountService:
     @staticmethod
     def create_account_by_username_and_password(*, params: CreateAccountByUsernameAndPasswordParams) -> Account:
         return AccountWriter.create_account_by_username_and_password(params=params)
 
+    @staticmethod
+    def create_or_update_notification_preferences(
+        *, account_id: str, preferences: NotificationPreferencesParams
+    ) -> NotificationPreferencesParams:
+        existing_preferences = AccountNotificationPreferenceReader.get_existing_notification_preferences_by_account_id(
```

Comment:
Create a method `create_or_update_notification_preferences` in `AccountNotificationPreferenceWriter` so the implementation in service class is simply a proxy